spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/StreamDeploymentController.java

```diff
@@ -247,6 +255,7 @@ private void deployStream(StreamDefinition stream, Map<String, String> streamDep
 			isDownStreamAppPartitioned = isPartitionedConsumer(appDeployTimeProperties,
 					upstreamAppSupportsPartition);
 
+			logger.info(String.format("Downloading resource URI [%s]", registration.getUri()));
 			Resource appResource = registration.getResource();
 			Resource metadataResource = registration.getMetadataResource();
```

Comment:
I also realised that we `prefetch` the metadata resources (during app registration) if there is an explicit metadata resource URI. Hence, we may not need the logging at all (in fact, there is no download at this point of time).